core/keyserver/validate.go

```diff
@@ -100,6 +103,34 @@ func validateListEntryHistoryRequest(in *pb.ListEntryHistoryRequest, currentRevi
 	return nil
 }
 
+// validateListUserRevisionsRequest checks the bounds on start and end revisions and returns an appropriate number of
+// revisions to return for this request.
+func validateListUserRevisionsRequest(in *pb.ListUserRevisionsRequest, offset, currentRevision int64) (int64, error) {
+	if in.StartRevision < 0 || in.StartRevision > currentRevision {
+		return 0, ErrInvalidStart
+	}
+	if in.EndRevision != 0 {
+		if in.EndRevision < in.StartRevision || in.EndRevision > currentRevision {
+			return 0, ErrInvalidEnd
+		}
+		currentRevision = in.EndRevision
+	}
+
+	revisions := int64(in.PageSize)
+	switch {
+	case in.PageSize < 0:
```

Comment:
it seems like you could pretty safely treat ==0 and <0 as the same case

Reply:
0 means the page size wasn't specified in the request so we can use a default, negative would be an invalid request.